kanidmd/src/lib/constants/schema.rs

```diff
@@ -56,7 +56,7 @@ pub const JSON_SCHEMA_ATTR_MAIL: &str = r#"
         "mail"
       ],
       "syntax": [
-        "UTF8STRING"
+        "EMAIL_ADDRESS"
```

Comment:
So the structure of schema is a whole thing. We store our schema in the database, so that in the future it can be replicated. Schema is something that admins can extend and customise, and having it in the database means we can make it consistent over the set of nodes in the topology.

Now of course, this has a problem. We have schema in the database - but we need schema to *query* that database in the first place.

So in schema.rs, we load an inmemory schema that is "just enough" to then process the database in an efficient manner (ie hit indexes) and then it is reloaded from the db/disk.

So these json structures are the DB schema elements, and we run them in a "migration stage" during server create and bootstrap. So these are used in the "initialise" functions in server.rs to load these into the database, and tthen we can use these for future migrations etc.